### ISSUE TITLE
TV show title banners won't download / are not appearing

### ISSUE THREAD

**Issue by GoogleCodeExporter** [peteaky/android-xbmcremote#312]
```
What steps will reproduce the problem?
1. Click the "Settings" button.
2. Click "Download Covers"
3. Click "TV Show Banners"

What is the expected output? What do you see instead?
The expected output is that it downloads all the TV show banners successfully.  
The actual output is that the Downloading TV Show Banners counter/status bar 
advances as if everything is going OK (in my case, eventually reaches 38/38 
banners), but then all I get is the message "0 posters downloaded".

And then when I click "Watch your TV shows", there are no banners.  Movie 
posters, music CD covers, TV season art, and TV episode screen caps all get 
pulled in fine, but TV title banners are NOT getting pulled in. All art is fine 
on the actual media center.


What version of XBMC Remote are you using?
0.8.2-beta1


Which Android device are you using (and which ROM, if custom)?
Motorola Droid - CyanogenMod 6.0


Which XBMC revision are you running and on which platform?
XBMC 9.11, Windows 7 x64



Please provide any additional information below.

```

Original issue reported on code.google.com by `scottfre...@gmail.com` on 19 Jul 2010 at 2:32

**Comment by GoogleCodeExporter** [comment]
```
Same issue.

HTC Hero
Stock 2.1 & 3 Different custom ROMs Fresh Darchdroid Aloysius
```

Original comment by `monkn...@gmail.com` on 19 Jul 2010 at 3:53

**Comment by GoogleCodeExporter** [comment]
```
Also got this issue, however it does end up getting some banners.
```

Original comment by `parfitt...@gmail.com` on 20 Jul 2010 at 6:55

Attachments:
- [tvbanners.jpg](https://storage.googleapis.com/google-code-attachments/android-xbmcremote/issue-312/comment-2/tvbanners.jpg)

**Comment by GoogleCodeExporter** [comment]
```
Thanks, seems definitely to be an issue.
```

Original comment by `phree...@gmail.com` on 20 Jul 2010 at 7:50
- Changed state: **Accepted**
- Added labels: **Milestone-Release0.8**, **Priority-Critical**
- Removed labels: **Priority-Medium**

**Comment by GoogleCodeExporter** [comment]
```
Issue 324 has been merged into this issue.
```

Original comment by `phree...@gmail.com` on 23 Jul 2010 at 8:11

**Comment by GoogleCodeExporter** [comment]
```
I have got this issue once. After checked the files on the sdcard I have seen 
that there is no image for the thumb in the "video/small" directory, but there 
is one with 0 length in the "video/medium" directory, so the application did 
not try to download it from the network.
```

Original comment by `Gabor.Fo...@gmail.com` on 27 Jul 2010 at 2:19

**Comment by GoogleCodeExporter** [comment]
```
Hi i am also experiencing this issue on my HTC Desire.
```

Original comment by `grant.fe...@gmail.com` on 29 Jul 2010 at 10:59

**Comment by GoogleCodeExporter** [comment]
```
Same issue here with HTC Desire, ROM: Open Desire 2.3a, Win 7 x64 9.11. I get 
the banners correct for the show, but none for the season nor for the actual 
episodes, except the last episode in each season which can play fine, but the 
other episodes in the season (which does not have thumbnails/info) can't play. 
```

Original comment by `axl.lars...@gmail.com` on 29 Jul 2010 at 12:06

**Comment by GoogleCodeExporter** [comment]
```
This code change solves my zero file length issue, however it's rather a cure 
than preventive medicine and probably only affective in that cases when a few 
thumbnails are missing.
```

Original comment by `Gabor.Fo...@gmail.com` on 29 Jul 2010 at 2:48

Attachments:
- [issue_312.patch](https://storage.googleapis.com/google-code-attachments/android-xbmcremote/issue-312/comment-8/issue_312.patch)

**Comment by GoogleCodeExporter** [comment]
```
Regarding my comment earlier #6, I have now found that if i connect my HTC 
Desire to an XBMC running on a windows 7 machine it works fine. I still cannot 
get any artwork when running XBMC on the old xbox. Is that by design??

Thanks
Grant.
```

Original comment by `grant.fe...@gmail.com` on 30 Jul 2010 at 2:26

**Comment by GoogleCodeExporter** [comment]
```
Same issue. Phone HTC Desire, Android 2.1. XBMC camelot running on appletv.
```

Original comment by `pedemeis...@gmail.com` on 30 Jul 2010 at 8:19

**Comment by GoogleCodeExporter** [comment]
```
I have had the same issue on My HTC Hero with Android 2.1 and HTC Hero with 
Android 2.1. On Unreleased Beta version. I just uninstalled Beta Version and 
Installed Version 0.8.3 Revision 686 From Android Market on both Devices and 
still same issue as above. I can not download TV Series Banners. 
```

Original comment by `Dnice...@gmail.com` on 4 Aug 2010 at 4:24

**Comment by GoogleCodeExporter** [comment]
```
Same issue on archos 5 internet tablet 32gig firmware 2.0.28 android 1.6
```

Original comment by `icanr...@gmail.com` on 9 Aug 2010 at 4:08

**Comment by GoogleCodeExporter** [comment]
```
This still occurs with the droid with android 2.2 on 0.8.3-beta (Rev 686) 
however it's fixed by erasing the offending banner from the following sections:

xbmc/(Section)/small
xbmc/(Section)/medium 
xbmc/(Section)/large

a more permanent fix would be to add a menu option that will refresh the TV 
series, album, etc thumb by erasing the old file, and then re-download the 
thumbnail, because that's the only way i can see to that will truly fix this 
issue.
```

Original comment by `kphilli...@gmail.com` on 20 Aug 2010 at 1:19

**Comment by GoogleCodeExporter** [comment]
```
i have this issue on 0.8.3 on HTC Evo, 2.2 fresh rom.   everything else 
downloads fine, but i have same issue as OP.  Looks like its downloading, but 
then  says 0 downloaded.
```

Original comment by `modelxpo...@gmail.com` on 22 Aug 2010 at 12:38

**Comment by GoogleCodeExporter** [comment]
```
Same problem, using xbox Revision: 30628 - and HTC desire 2.2...
```

Original comment by `r...@szimnau.dk` on 15 Sep 2010 at 4:50

**Comment by GoogleCodeExporter** [comment]
```
Downloaded Newest Update from Market v0.8.4-beta1 Revision 703. Still same 
issue as above Progress bar continues until 100% then message pops up " 0 
posters downloaded." Same issue with Revision 686. Both Devices HTC Hero 
running Android 2.1 and HTC EVO Running Android 2.2
```

Original comment by `Dnice...@gmail.com` on 15 Oct 2010 at 2:50

**Comment by GoogleCodeExporter** [comment]
```
Could you help us by attaching a log? There is an instruction of how to on the 
http://forum.xbmc.org/showthread.php?t=70243 paragraph #16
Thanks
```

Original comment by `Gabor.Fo...@gmail.com` on 15 Oct 2010 at 1:20

**Comment by GoogleCodeExporter** [comment]
```
What helped for me was looking in het log. I saw the app was downloading 
/path_to_series/banner.jpg (after exporting the library in seperate files and 
thumbnails). But in all my folders I saw /path_to_series_/folder.jpg. So I 
renamed them all to banner.jpg and now nearly all the banners are downloaded, 
except the one I recently added and haven't exported yet.

Maybe someone is helped with this solution.
```

Original comment by `xxf...@gmail.com` on 15 Oct 2010 at 2:50

**Comment by GoogleCodeExporter** [comment]
```
I'm guessing this may stem from the same issue, so I figured I'd just add a 
comment here. I'm noticing that none of my Season 0 banners DLed to my Android 
device either.
```

Original comment by `XxSerber...@gmail.com` on 4 Jan 2011 at 2:06

**Comment by GoogleCodeExporter** [comment]
```
I have 2 Series Harddrives one ist local in the xbmc and one ist mounted with 
smb from my nasbox in xbmc. All Banners from the series on the local drive will 
be shown but non from the mounted drive. In Xbmc all banners from both drives 
will be shown.
i am using the newest version from the market. 
```

Original comment by `Rentner....@googlemail.com` on 8 Feb 2011 at 7:10

**Comment by GoogleCodeExporter** [comment]
```
It tries to download banner.jpg for those who have posters or fanart as folder 
icon, then it goes for the show name. So if you see banners in XBMC and you 
don't have a banner.jpg, the list should still show banners.

Will discuss with the team how we could offer better icon support for different 
formats in JSON-RPC. I don't think I'll invest a lot of time with HTTPAPI in 
this.
```

Original comment by `phree...@gmail.com` on 20 Feb 2011 at 11:02

**Comment by GoogleCodeExporter** [comment]
```
Same issue here. HTC Desire, ATV2, Stora network drive. Movie thumbs, album 
artwork, fanart etc all no issue.
```

Original comment by `rjd...@gmail.com` on 8 Mar 2011 at 10:43

**Comment by GoogleCodeExporter** [comment]
```
same problem on my Samsung galaxy S + Apple TV2.
He manages to only download 1 tv banner, the one i created myself.
```

Original comment by `kanah...@gmail.com` on 2 Apr 2011 at 9:29

**Comment by GoogleCodeExporter** [comment]
```
The problem I saw was it attempted to download the file using one method 
(expecting a base64 response or nothing if it fails) but it returns HTML, I 
just added a check to that so it looks for < (which is invalid in base64) and 
if it is found considers that a failure as well.  This fixed it for me at least.

Index: XBMC Remote/src/org/xbmc/httpapi/client/Client.java
===================================================================
--- XBMC Remote/src/org/xbmc/httpapi/client/Client.java (revision 762)
+++ XBMC Remote/src/org/xbmc/httpapi/client/Client.java (working copy)
@@ -106,7 +106,7 @@
                size = size < ThumbSize.BIG ? ThumbSize.MEDIUM : ThumbSize.BIG;
                try {
                        String b64enc = mConnection.query("FileDownload", url, manager);
-                       if (b64enc.length() <= 0) {
+                       if ( (b64enc.length() <= 0) || (b64enc.indexOf('<') != 
-1) ) {
                                if (fallbackUrl != null) {
                                        Log.i(TAG, "*** Downloaded cover has size null, retrying with fallback:");
                                        b64enc = mConnection.query("FileDownload", fallbackUrl, manager);
```

Original comment by `koshatul` on 8 Apr 2011 at 7:26

**Comment by GoogleCodeExporter** [comment]
```
Also, I had a cache full of previously retrieved banners, sine new ones were 
failing to download I cleared the cache, then none of them would download.
```

Original comment by `koshatul` on 8 Apr 2011 at 7:28

**Comment by GoogleCodeExporter** [comment]
```
This is an issue for me as well.  I just upgraded phones, and it work perfectly 
on my Samsung Galaxy S.  But now on my Nexus S, i cannot get any TV posters to 
download.  Hell im having multiple issues with XBMC remote on the Nexus S, but 
thats for another ticket
```

Original comment by `wyi...@gmail.com` on 16 Apr 2011 at 2:12

**Comment by GoogleCodeExporter** [comment]
```
Just an update...All movie posters download fine.  I cleared the cache and that 
didnt help.  

HTPC: Starting XBMC, Platform: Linux (Ubuntu 10.04.2 LTS, 2.6.32-29-generic 
i686). Built on Mar  9 2011 (Git:Unknown)  
XBMC 10.1 Git:Unknown

Phone: Google/Samsung Nexus S
ROM: CyanogenMod 7

```

Original comment by `wyi...@gmail.com` on 16 Apr 2011 at 2:19

**Comment by GoogleCodeExporter** [comment]
```
This affects me too. V0.8.6-beta1 on a nexus one (2.3) with xbmc dharma. Movies 
download fine, but TV banners just say 0 Downloaded after the progress bar 
fills.
```

Original comment by `marcbo...@gmail.com` on 28 Apr 2011 at 2:40

**Comment by GoogleCodeExporter** [comment]
```
I have the same Problem on a Samsung Galaxy S i9000 (Android 2.2.1).

I have downloaded and deleted the banners several times. Every time i tried 
this other banners were transferred successfully, but never all banners. My 
files are on a NAS and every TV show has a "banner.jpg" in its folder. Those 
banners are shown correctly within XBMC.

Seems like the download isn't stable and you have to be lucky to get a banner. 
I could probably try to delete the files that weren't transferred correctly and 
redownload them, but that's rather difficult, because the filenames consist of 
mixed numbers and letters.
```

Original comment by `manuel.p...@googlemail.com` on 10 May 2011 at 6:32

**Comment by GoogleCodeExporter** [comment]
```
dame here!

```

Original comment by `muttley...@gmail.com` on 26 May 2011 at 11:34

**Comment by GoogleCodeExporter** [comment]
```
I am facing the same issue as well. I am able to view the banners and info all 
my Movies but not my TV Shows. I am running it on a Droid X with gingerbread.
```

Original comment by `bramesh....@gmail.com` on 9 Jul 2011 at 7:31

**Comment by GoogleCodeExporter** [comment]
```
Is there an official way to submit a patch ?
```

Original comment by `koshatul` on 11 Jul 2011 at 6:48

**Comment by GoogleCodeExporter** [comment]
```
Same issue on the Samsung Galaxy S2 phone.
Would never download all banners, out of 35 TV series I would be lucky to get 3 
or 4.

Everytime I clear the cache and try downloading the banners again, it would 
pick up some other banners that wasn't picked up previously. 

Frustrations!!!!! 
```

Original comment by `phey...@gmail.com` on 20 Jul 2011 at 1:18

**Comment by GoogleCodeExporter** [comment]
```
One thing you might want to try is keep pressing the download TV banner button 
over and over and over and over again. It will eventually download 1 at a 
time......

Then exit the program completely from your phone, and reconnect to 
server....then the banners will start showing.
```

Original comment by `phey...@gmail.com` on 20 Jul 2011 at 1:25

**Comment by GoogleCodeExporter** [comment]
```
I applied my patch and compiled it locally, works fine for me, on both nexus 
one and sgs2
```

Original comment by `koshatul` on 22 Jul 2011 at 2:27

**Comment by GoogleCodeExporter** [comment]
```
What is the HTML page about that you avoid with this patch?
```

Original comment by `Gabor.Fo...@gmail.com` on 23 Jul 2011 at 12:37

**Comment by GoogleCodeExporter** [comment]
```
The original code thinks that a 0 length return is a fail and falls over to the 
newer version request, but in the latest xbmc it returns a HTML error page, so 
if I match the < since it is an invalid base64 character it considers that a 
fail as well, it could probably be done much better.
```

Original comment by `koshatul` on 25 Jul 2011 at 9:18

**Comment by GoogleCodeExporter** [comment]
```
will confirm same problem for me. Running Synergy RC2 on EVO, Android XBMC 
v0.8.6-beta1 r768.

TLDR: android app only looks for "banner.jpg" and not "folder.jpg". Needs to be 
modified to look for either.

Long Story:
Confirming Comment #18. All of my TV series banners are in the format of 
"folder.jpg" (done through Media Companion scraping). Having Android XBMC 
Remote download TV Banner Covers results is it going through ##/## completely, 
and then "0 posters downloaded". When I changed one series file name from 
"folder.jpg" to "banner.jpg" and redownloaded TV Banner Covers, it went through 
##/## completely, then said "1 poster downloaded", which I confirmed by 
browsing TV series and seeing it.
```

Original comment by `Zxur...@gmail.com` on 25 Jul 2011 at 9:23

**Comment by GoogleCodeExporter** [comment]
```
Hasn't koshatul determined the issue here? How can we get the patch added?
```

Original comment by `colinjon...@gmail.com` on 4 Aug 2011 at 6:36

**Comment by GoogleCodeExporter** [comment]
```
I'm also having this problem and noticed that the problem occurs only with 
banners/posters I have changed from the default one on my XBMC machine.  The 
ones that I have left as the images that appeared originally all download to my 
phone fine.

I would like to suggest that an option to long-press any missing/existing image 
and have the option to change it - this would help get rid of those annoying 
grey TV screens for "Season 0" of a show (if I should suggest this as a 
separate feature, please let me know but I think it would be a good 
introduction and would best be implemented as the issues mentioned above are 
being dealt with)

Using;
Phone : HTC Desire 
Phone OS : Android 2.2 (with HTC Sence UI)
XBMC : 10.0.35647
XBMC Remote : 0.8.6-beta1 (rev768)
```

Original comment by `daveho...@gmail.com` on 19 Aug 2011 at 11:03

**Comment by GoogleCodeExporter** [comment]
```
Seeing the same thing as in Comment #2 from well over a year ago on both an 
Asus Transformer (Honeycomb 3.2) and HTC Evo (Gingerbread 2.3.5).  It seems 
completely random which show it refuses to download each time I clear the cache 
and re-download.  The same show that doesn't get grabbed one time may very well 
get pulled the next.
```

Original comment by `jdrus...@gmail.com` on 13 Sep 2011 at 7:55

**Comment by GoogleCodeExporter** [comment]
```
I have the same issue, with done banners only. One thing I realised is that the 
banners only don't appear on TV series which are shared over SMB. Those which 
are stored locally on the same server as xbmc show up fine, but those which are 
stored separately and accessed via SMB do not. 

Xbmc is installed on Ubuntu, SMB server is windows. Phone is desire s, desire 
z, legend, all did the same. 
```

Original comment by `mirob...@gmail.com` on 30 Nov 2011 at 9:26

**Comment by GoogleCodeExporter** [comment]
```
I have the same issue.  Movie posters download fine, but no TV Banners or 
Episode Images will download.  I let it run for a long time downloading over 3k 
Episode images, then it said "0 posters downloaded".  All of my banners and 
posters are stored on a mounted SMB folder, but there is no way that my phone 
could know that.  I am running XBMC in English.

XBMC Version: XBMC Live (Dharma)
Phone: HTC Evo running CyanogenMod 7
XBMC Remote: v0.8.8-beta1.

Someone in a prior post mentioned a location on the phone where these images 
should be stored, but I cannot find that location.
```

Original comment by `themadwi...@gmail.com` on 17 Mar 2012 at 7:01

**Comment by GoogleCodeExporter** [comment]
```
Still having this issue with 0.8.8beta1 on Galaxy S running MIUI V4
```

Original comment by `ylverh...@gmail.com` on 14 May 2012 at 3:58

**Comment by GoogleCodeExporter** [comment]
```
I am seeing this issue on Asus transformer prime ICS. Some banners display but 
the majority do not, i had previously used this tablet with a windows 7 build 
withou any issues at all. Now my new build is ubuntu using samba storage. 
Please let me know if i can help in any way, the remote is still functional but 
the banners give a nice fuzzy feeling :)

Many thanks

```

Original comment by `ItalianP...@gmail.com` on 5 Jun 2012 at 7:29

**Comment by GoogleCodeExporter** [comment]
```
To follow on from the comment above about locally stored TV series working and 
those shared over SMB not working. I have just configured by NAS to allow NFS 
connections from the Ubuntu system running XMBC and added the source in again 
using a local mount. I then removed the host from the XBMC remote and re-added 
it - now the banners work correctly. In my case it looks like an 
incompatibility between Samba shares and Ubuntu. Hope this helps others if you 
have the option of running NFS alongside Samba.


```

Original comment by `ItalianP...@gmail.com` on 5 Jun 2012 at 10:28
